lib/web_ui/lib/src/engine/window.dart

```diff
@@ -198,6 +203,54 @@ class EngineWindow extends ui.Window {
       callback(data);
     });
   }
+
+  /// The setting indicating the current brightness mode of the host platform.
+  /// If the platform has no preference, [platformBrightness] defaults to [Brightness.light].
+  ui.Brightness get platformBrightness => _platformBrightness;
+  ui.Brightness _platformBrightness = ui.Brightness.light;
+
+  /// Updates [_platformBrightness] and invokes [onPlatformBrightnessChanged]
+  /// callback if [_platformBrightness] changed.
+  void _updatePlatformBrightness(ui.Brightness newPlatformBrightness) {
+    ui.Brightness previousPlatformBrightness = _platformBrightness;
+    _platformBrightness = newPlatformBrightness;
+
+    if (previousPlatformBrightness != _platformBrightness &&
+        onPlatformBrightnessChanged != null)
+      onPlatformBrightnessChanged();
+  }
+
+  /// Reference to css media query that indicates the user theme preference on the web.
+  final html.MediaQueryList _brightnessMediaQuery = html.window.matchMedia('(prefers-color-scheme: dark)');
+
+  /// A callback that is invoked whenever [_brightnessMediaQuery] changes value.
+  ///
+  /// Updates the [_platformBrightness] with the new user preference.
+  html.EventListener _brightnessMediaQueryListener;
+
+  /// Set the callback function for listening changes in [_brightnessMediaQuery] value.
+  void _addBrightnessMediaQueryListener() {
+    _updatePlatformBrightness(_brightnessMediaQuery.matches ? ui.Brightness.dark : ui.Brightness.light);
+
+    _brightnessMediaQueryListener = (html.Event event) {
+      final html.MediaQueryListEvent mqEvent = event;
+      _updatePlatformBrightness(mqEvent.matches ? ui.Brightness.dark : ui.Brightness.light);
+    };
+    _brightnessMediaQuery.addListener(_brightnessMediaQueryListener);
```

Comment:
This might leak after a hot reload. cc @yjbanov ?

Reply:
Yep. Let's use `registerHotRestartListener` to remove the listener. Here's an example: https://github.com/flutter/engine/blob/master/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1077